src/main/java/com/synopsys/arc/jenkinsci/plugins/jobrestrictions/restrictions/job/StartedByMemberOfGroupRestriction.java

```diff
@@ -0,0 +1,193 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+import jenkins.model.Jenkins;
+
+import org.acegisecurity.GrantedAuthority;
+import org.acegisecurity.userdetails.UserDetails;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import com.synopsys.arc.jenkinsci.plugins.jobrestrictions.Messages;
+import com.synopsys.arc.jenkinsci.plugins.jobrestrictions.restrictions.JobRestriction;
+import com.synopsys.arc.jenkinsci.plugins.jobrestrictions.restrictions.JobRestrictionDescriptor;
+import com.synopsys.arc.jenkinsci.plugins.jobrestrictions.util.GroupSelector;
+
+/**
+ *
+ * @author Oleg Nenashev <o.v.nenashev@gmail.com>
+ * @since 0.4
+ */
+public class StartedByMemberOfGroupRestriction extends JobRestriction {
+
+	private final List<GroupSelector> groupList;
+	private final boolean checkGroupsFromUpstreamProjects;
+
+	private transient Set<String> acceptedGroups = null;
+
+	@DataBoundConstructor
+	public StartedByMemberOfGroupRestriction(List<GroupSelector> groupList,
+			boolean checkGroupsFromUpstreamProjects) {
+		this.groupList = groupList;
+		this.checkGroupsFromUpstreamProjects = checkGroupsFromUpstreamProjects;
+	}
+
+	public List<GroupSelector> getGroupList() {
+		return groupList;
+	}
+
+	public boolean isCheckGroupsFromUpstreamProjects() {
+		return checkGroupsFromUpstreamProjects;
+	}
+
+	private synchronized @Nonnull
+	Set<String> getAcceptedGroups() {
+		if (acceptedGroups == null) {
+			final List<GroupSelector> selectors = getGroupList();
+			acceptedGroups = new HashSet<String>(selectors.size());
+			for (GroupSelector selector : selectors) {
+				acceptedGroups.add(selector.getSelectedGroupId()); // merge
+																	// equal
+																	// entries
+			}
+		}
+		return acceptedGroups;
+	}
+
+	private boolean acceptsUser(@CheckForNull String userId) {
+		if (userId == null) {
+			return false;
+		}
+		SecurityRealm sr = Jenkins.getInstance().getSecurityRealm();
+		UserDetails userDetails = sr.loadUserByUsername(userId);
+		for (String groupId : getAcceptedGroups()) {
+			GrantedAuthority[] authorities = userDetails.getAuthorities();
+			for (GrantedAuthority auth : authorities) {
+				String authority = auth.getAuthority();
+				if (authority.equals(groupId)) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/** package */
+	boolean canTake(@Nonnull List<Cause> causes) {
+		boolean userIdCause = false;
+		boolean rebuildCause = false;
+		boolean upstreamCause = false;
+		boolean aUserIdWasNotAccepted =false;
+		boolean userIdCauseExists=false;;
+		for (Cause cause : causes) {
+			if (cause.getClass().equals(Cause.UserIdCause.class) && !aUserIdWasNotAccepted) {
+				userIdCauseExists=true;
+				//if several userIdCauses exists, be defensive and don't allow if one is not accepted.
+				final @CheckForNull
+				String startedBy = ((Cause.UserIdCause) cause).getUserId();
+				if(acceptsUser(startedBy) )
+				{
```

Comment:
I would re-work the formatting to satisfy common Jenkins styles (Sun/Oracle codestyle).
BTW, I'm going to generalize the method after the merge in order to avoid code dups between user and group restrictions, so you can just leave it to me